### PR TITLE
Slide in

### PR DESCRIPTION
**Pull request by florianGierlichs**

### CODE REVIEW

**Comment by aweigl** [comment]
Wow great work!

**Review by rokoc003 — APPROVED** [review]
Woooow good job! 🎉 😍

File: index.html [hunk 1]
```diff
@@ -11,28 +11,60 @@ <h1>My first Calculator</h1>
...
-        <button class="calculator__input__spec">*</button>
-        <button class="calculator__inputs calculator__num">1</button>
-        <button class="calculator__inputs calculator__num">2</button>
-        <button class="calculator__inputs calculator__num">3</button>
-        <button class="calculator__input__spec">-</button>
-        <button class="calculator__inputs">.</button>
-        <button class="calculator__inputs calculator__num">0</button>
-        <button class="calculator__input__result">=</button>
-        <button class="calculator__input__spec">+</button>
+        <div class="magic__wrapper">
```
Comment (lmachens) [inline index.html]: I think it's not necessary to create a wrapper element. instead, you could apply the css of `magic__wrapper` to the buttons

File: src/partials/_calculator.scss [hunk 2]
```diff
@@ -120,26 +165,62 @@ h1 {
...
+// }
+
+.calculation {
+  font-size: 2rem;
+  background-color: springgreen;
+}
+
+.active {
+  position: relative;
+  right: 0px !important;
```
Comment (lmachens) [inline src/partials/_calculator.scss]: `!important` is not a good practice. can you think about a different solution?

File: src/partials/_calculator.scss [hunk 3]
```diff
@@ -43,13 +44,25 @@ h1 {
   padding: 12px;
   top: -900px;
   position: relative;
+
+  transition: all 2s ease-in-out;
+}
+
+.calcActive {
```
Comment (lmachens) [inline src/partials/_calculator.scss]: the naming convention is different from your other class names.
you are using BEM with dash seperated names. Here, you are using camelCase.
Try not to mix conventions.

File: src/calculator.js [hunk 4]
```diff
@@ -14,11 +14,39 @@ let numberOne = "";
...
+  const listInputs = document.querySelectorAll(".calculator__inputs");
+  const listSpec = document.querySelectorAll(".calculator__input__spec");
+  const listTotal = [...listInputs, ...listSpec]; // new Array()
+
+  const equalSign = document.querySelector(".calculator__input__result");
+
+  equalSign.style.left;
+
+  listTotal.forEach(item => {
+    const randNum = Math.floor(Math.random() * (1000 - 700 + 1) + 700);
```
Comment (lmachens) [inline src/calculator.js]: 👍 for creating random numbers

File: src/calculator.js [hunk 5]
```diff
@@ -14,11 +14,39 @@ let numberOne = "";
...
 
+const randomizeShit = () => {
+  //do work
+  const listInputs = document.querySelectorAll(".calculator__inputs");
+  const listSpec = document.querySelectorAll(".calculator__input__spec");
+  const listTotal = [...listInputs, ...listSpec]; // new Array()
+
+  const equalSign = document.querySelector(".calculator__input__result");
+
+  equalSign.style.left;
```
Comment (lmachens) [inline src/calculator.js]: This line doesn't do anything

**Review by lmachens — COMMENTED** [review]
Looks like you have a lot of fun 😄 
Good job!

File: index.html (hunk 1 above)
Comment (florianGierlichs) [reply index.html]: The buttons need to have `position: relative;` to move around. In this case i get issues with my flexbox around the buttons. My solution was, to put a container around every button, which is then related to the flexbox around. so the button in the end can be `position: relative;` and fly around, while their parent (magic__wrapper) will always on the final-position. So far i couldn't find another solution.

File: src/partials/_calculator.scss (hunk 2 above)
Comment (florianGierlichs) [reply src/partials/_calculator.scss]: That's right and i'm not happy with this solution either. But my css properties got overwritten in a way i don't want it, so like the comment above, i couldn't find another solution yet.

**Pull request merged**